pkg/cli/admin/catalog/build.go

```diff
@@ -0,0 +1,216 @@
+package catalog
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/operator-framework/operator-registry/pkg/apprclient"
+	"github.com/operator-framework/operator-registry/pkg/sqlite"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/klog"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/util/templates"
+
+	imgappend "github.com/openshift/oc/pkg/cli/image/append"
+)
+
+func NewBuildImageOptions(streams genericclioptions.IOStreams) *BuildImageOptions {
+	return &BuildImageOptions{
+		IOStreams:           streams,
+		AppRegistryEndpoint: "https://quay.io/cnr",
+		From:                "quay.io/operator-framework/operator-registry-server:latest",
```

Comment:
Do we need to default that image name here? Have a look at how `oc adm must-gather` resolves image from cluster's internal image stream: https://github.com/openshift/oc/blob/8fdb79e549651c0f3c91d54349715309b5d149d3/pkg/cli/admin/mustgather/mustgather.go#L102-L107 and only if that fails defaults to quay. I think it'd be a better option than hardcoding this.